specs/alerting-api.adoc

```diff
@@ -0,0 +1,360 @@
+
+----
+
+# flake8: noqa
+data = """---
+namespace.tendrl.alerting:
+  objects:
+    Config:
+      value: '/_tendrl/config/alerting'
+      data:
+        help: "The configurations path"
+        type: json
+      enabled: true
+    Alert:
+      attrs:
+        alert_id:
+          help: 'The unique identifier of alert'
+          type: String
+        node_id:
+          help: 'The unique identifier of node on which alert was detected'
+          type: String
+        time_stamp:
+          help: 'The timestamp at which alert was observed'
+          type: String
+        resource:
+          help: 'The resource with problem for which alert was raised'
+          type: String
+        current_value:
+          help: 'The current magnitude(status/utilization) of problem'
+          type: String
+        tags:
+          help: 'Alert specific fields that cannot be generalized for all alerts'
+          type: json
+        alert_type:
+          help: 'The type(status/percentage utilization) of alert'
+          type: String
+        severity:
+          help: 'The severity of alert'
+          type: String
+        significance:
+          help: 'The significance of notifying alert'
+          type: String
+        ackedby:
+          help: 'Entity/person acking the alert'
+          type: String
+        acked:
+          help: 'Indication of whether alert is acked or not'
+          type: Boolean
+        pid:
+          help: 'The id of process raising the alert'
+          type: String
+        source:
+          help: 'The process raising the alert'
+          type: String
+      enabled: true
+      value: alerts/$Alert.alert_id
+      list: alerts/
+    NotificationMedia:
+      attrs:
+        value:
+          help: 'The list of supported notification medium'
+          type: String Array
+      list: alerting/notification_medium/supported/
+    AlertTypes:
+      attrs:
+        module-name:
+          help: 'The name of the tendrl module'
+        value:
+          help: 'The list of supported types of alerts corresponding to module-name
+          type: String array
+      value: alerting/alert_types/$AlertTypes.module-name
+      list: alerting/alert_types/
+    NotificationConfig:
+      attrs:
+        id:
+          help: 'A unique identifier of the config'
+          type: String
+        value:
+          help: 'The validated configuration corresponding to notification medium'
+          type: json
+      value: alerting/notification_medium/$NotificationMedia.value[i]/config/$NotificationConfig.id
+      list: alerting/notification_medium/$NotificationMedia.value[i]/config/
+tendrl_schema_version: 0.3
+"""
+
+----
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The following apis need to be exposed by the tendrl-api module:
+
+* This adds the following apis:
+  ** Api to get list of currently supported means of notification.
+
+----
+GET /alerting/notification_medium/supported
```

Comment:
Do we need api to be `/alerting/notification_medium/supported` or `/alerting/notification_medium` is enough ?